### PR TITLE
Use libblockdev to get LUKS UUID for LUKS open

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Because it is more reliable than using from udev.

### CODE REVIEW

File: src/udiskslinuxblock.c [hunk 1]
```diff
@@ -1530,14 +1530,16 @@ has_whitespace (const gchar *s)
 }
```
Comment (vpodzime) [inline src/udiskslinuxblock.c]: "...than getting data from udev"

File: src/udiskslinuxblock.c [hunk 2]
```diff
@@ -1530,14 +1530,16 @@ has_whitespace (const gchar *s)
...
-    return g_strdup_printf ("luks-%s", uuid);
+  gchar *uuid = bd_crypto_luks_uuid (udisks_block_get_device (block), error);
 
-  return NULL;
+  if (!uuid)
+    return NULL;
+  else
+    {
+      return g_strdup_printf ("luks-%s", uuid);
+    }
```
Comment (vpodzime) [inline src/udiskslinuxblock.c]: Extra braces here. And maybe switching the branches here (together with the condition of course) would make more sense?

File: src/udiskslinuxblock.c (hunk 1 above)
Comment (vojtechtrefny) [inline src/udiskslinuxblock.c]: Fixed, thanks.

File: src/udiskslinuxblock.c (hunk 2 above)
Comment (vojtechtrefny) [inline src/udiskslinuxblock.c]: Fixed.

**Pull request merged**